tmt/steps/execute/internal.py

```diff
@@ -322,48 +258,50 @@ def go(self, guest):
     def _run_tests(self, guest, extra_environment=None):
         """ Execute tests on provided guest """
         extra_environment = extra_environment or {}
+
         # Prepare tests and helper scripts, check options
         tests = self.prepare_tests()
         exit_first = self.get('exit-first', default=False)
-        self.step.write(FILE_SUBMIT_NAME, FILE_SUBMIT_SCRIPT)
-
-        with self._setup_reboot(guest):
-            # Push workdir to guest and execute tests
-            guest.push()
-            index = 0
-            while index < len(tests):
-                test = tests[index]
-                if not hasattr(test, "_reboot_count"):
-                    test._reboot_count = 0
-                self.execute(
-                    test, guest, progress=f"{index + 1}/{len(tests)}",
-                    extra_environment=extra_environment)
-
-                # Pull test logs from the guest, exclude beakerlib backups
-                if test.framework == "beakerlib":
-                    exclude = [
-                        "--exclude",
-                        self.data_path(test, "backup*", full=True)]
-                else:
-                    exclude = None
-                guest.pull(
-                    source=self.data_path(test, full=True),
-                    extend_options=exclude)
-
-                # Handle reboot, check results
-                if self._handle_reboot(test, guest):
-                    continue
-                self._results.append(self.check(test))
-                if (exit_first and
-                        self._results[-1].result not in ('pass', 'info')):
-                    # Clear the progress bar before outputting
-                    self._show_progress('', '', True)
-                    self.warn(
-                        f'Test {test.name} failed, stopping execution.')
-                    break
-                index += 1
-            # Overwrite the progress bar, the test data is irrelevant
-            self._show_progress('', '', True)
+
+        # Prepare scripts
+        if not isinstance(guest, GuestLocal):
+            self.prepare_scripts(guest)
```

Comment:
Same as my comment WRT `GuestContainer` - let's add a flag rather than requiring `internal` executor to be aware of guest internals. I suppose here we don't want to overwrite files on my laptop when using `local` provisioning plugin, there sure can be a flag to control this behavior.

If I decide the default `local` is not good enough for me, and I come up with my custom subclass, this "do not mess with scripts on user's laptop" feature would be completely closed to me, because this `isinstance()` check wouldn't recognize my `GuestLocalButMuchBetter` class.

Reply:
Yeah, this is not my change, it was there before, but yeah, should be part of the refactoring ... thanks

Reply:
added a localhost flag to indicate the guest needs special handling.